File: 021/main.js
/**
 * @param {ListNode} l1
 * @param {ListNode} l2
 * @return {ListNode}
 */

function ListNode (val) {
  this.val = val
  this.next = null
}

var mergeTwoLists = function (l1, l2) {
  let arr = []
  while (l1 != null) {
    arr.push(l1.val)
    l1 = l1.next
  }
  while (l2 != null) {
    arr.push(l2.val)
    l2 = l2.next
  }
  if (arr.length === 0) {
    return null
  }
  arr.sort((a, b) => {
    return a - b > 0 ? 1 : -1
  })
  let res = new ListNode(-1)
  let doc = res
  for (let i = 0; i < arr.length; i++) {
    doc.val = arr[i]
    if (i < arr.length - 1) {
      doc.next = new ListNode(-1)
      doc = doc.next
    }
  }
  return res
}

let l1 = {
  val: 2,
  next: {
    val: 4,
    next: {
      val: 3,
      next: null
    }
  }
}

let l2 = {
  val: 5,
  next: {
    val: 6,
    next: {
      val: 4,
      next: null
    }
  }
}

console.log(mergeTwoLists(l1, l2))

File: 021/main.cpp
#include <iostream>
#include <algorithm>
#include <vector>
using namespace std;

struct ListNode {
    int val;
    ListNode *next;
    ListNode(int x) : val(x), next(NULL) {}
};

class Solution {
public:
    ListNode* mergeTwoLists(ListNode* l1, ListNode* l2) {
        vector<int> arr;
        while (l1) {
            arr.push_back(l1->val);
            l1 = l1->next;
        }
        while (l2) {
            arr.push_back(l2->val);
            l2 = l2->next;
        }
        sort(arr.begin(), arr.end());
        ListNode* res = new ListNode(-1);
        ListNode* ans = res;
        for (int i = 0; i < arr.size(); i++) {
            ListNode* tmp = new ListNode(-1);
            tmp->val = arr[i];
            ans->next = tmp;
            ans = ans->next;
        }
        return res->next;
    }
};

int main()
{
    Solution sol;
    ListNode* l1 = new ListNode(1);
    ListNode* l12 = new ListNode(2);
    ListNode* l13 = new ListNode(4);
    ListNode* l2 = new ListNode(1);
    ListNode* l22 = new ListNode(3);
    ListNode* l23 = new ListNode(4);
    l12->next = l13;
    l1->next = l12;
    l22->next = l23;
    l2->next = l22;
    ListNode* res = sol.mergeTwoLists(l1, l2);
    while (res) {
        cout << res->val << " ";
        res = res->next;
    }
    return 0;
}
